pyuvdata/uvflag/tests/test_uvflag.py

```diff
@@ -648,6 +798,294 @@ def test_read_write_loop_missing_shapes(uvdata_obj, test_outfile):
+            None,
+        ),
+        (
+            "baseline",
+            ["antenna_names", "antenna_numbers", "antenna_positions"],
+            UserWarning,
+            "Nants_telescope, antenna_names, antenna_numbers, antenna_positions are "
+            "not set or being overwritten. Using known values for HERA.",
+            "reset_telescope_params",
+        ),
+        (
+            "baseline",
+            ["antenna_numbers"],
+            [UserWarning, UserWarning, DeprecationWarning],
+            [
+                "antenna_numbers is not set but cannot be set using known values for "
+                "HERA because the expected shapes don't match.",
+                "antenna_numbers not in file, cannot be set based on ant_1_array and "
+                "ant_2_array because Nants_telescope is greater than Nants_data.",
+                "The antenna_numbers is not set. It will be a required parameter "
+                "starting in pyuvdata version 2.4",
+            ],
+            None,
+        ),
+        (
+            "baseline",
+            ["antenna_names"],
+            UserWarning,
+            [
+                "antenna_names is not set but cannot be set using known values for "
+                "HERA because the expected shapes don't match.",
+                "antenna_names not in file, setting based on antenna_numbers",
+            ],
+            None,
+        ),
+        (
+            "baseline",
+            ["antenna_numbers"],
+            UserWarning,
+            [
+                "antenna_numbers is not set but cannot be set using known values for "
+                "HERA because the expected shapes don't match.",
+                "antenna_numbers not in file, setting based on ant_1_array and "
+                "ant_2_array.",
+            ],
+            "remove_extra_metadata",
+        ),
+        (
+            "antenna",
+            ["antenna_numbers"],
+            [UserWarning, UserWarning, DeprecationWarning],
+            [
+                "antenna_numbers is not set but cannot be set using known values for "
+                "HERA because the expected shapes don't match.",
+                "antenna_numbers not in file, cannot be set based on ant_array because "
+                "Nants_telescope is greater than Nants_data.",
+                "The antenna_numbers is not set. It will be a required parameter "
+                "starting in pyuvdata version 2.4",
+            ],
+            None,
+        ),
+        (
+            "antenna",
+            ["antenna_numbers"],
+            [UserWarning, UserWarning],
+            [
+                "antenna_numbers is not set but cannot be set using known values for "
+                "HERA because the expected shapes don't match.",
+                "antenna_numbers not in file, setting based on ant_array.",
+            ],
+            "remove_extra_metadata",
+        ),
+    ],
+)
+def test_read_write_loop_missing_telescope_info(
```

Comment:
took me a few reads but this monstrosity of a test function is neat. I didn't get why we needed to set the things you deleted but it only gets deleted out of the hdf5 file and still exists on the original object.

Reply:
:joy: yeah, I was pleased to be able to put all these cases in the parametrize but it is hard to grock.